src/main/java/com/kmorarash/hwJDBC/validator/UserValidator.java

```diff
@@ -0,0 +1,31 @@
+package com.kmorarash.hwJDBC.validator;
+
+
+import com.kmorarash.hwJDBC.dto.UserRegistrationDto;
+
+import java.util.regex.Pattern;
+
+public class UserValidator {
+    private static final Pattern EMAIL_PATTERN = Pattern.compile("^[\\w-\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$");
+    private static final Pattern PHONE_PATTERN = Pattern.compile("^\\+?\\d{10,15}$");
+
+    public void validate(UserRegistrationDto userRegistrationDto) {
+        if (userRegistrationDto.getEmail() == null || userRegistrationDto.getEmail().isEmpty()) {
+            throw new RuntimeException("Email is required.");
+        }
+
+        if (!EMAIL_PATTERN.matcher(userRegistrationDto.getEmail()).matches()) {
+            throw new RuntimeException("Invalid email format.");
```

Comment:
It is better to throw more specific exception
```suggestion
            throw new ValidationException("Invalid email format.");
```